arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -381,9 +382,13 @@ public VectorHolder read(VectorHolder reuse, int numValsToRead) {
         for (int i = 0; i < numValsToRead; i += 1) {
           vec.getDataBuffer().setLong(i * Long.BYTES, rowStart + i);
         }
+        for (int i = 0; i < numValsToRead; i += 1) {
+          BitVectorHelper.setValidityBitToOne(vec.getValidityBuffer(), i);
+        }
         nulls = new NullabilityHolder(numValsToRead);
       }
 
+      rowStart += numValsToRead;
```

Comment:
Looks like this was an oversight in the original PR. FYI @chenjunjiedada.

Reply:
Great catch! Let me update the unit test as well.